en/lessons/specifics/ecto_basics.md

```diff
@@ -0,0 +1,196 @@
+
+```elixir
+defmodule Friends.Repo do
+  use Ecto.Repo, otp_app: :friends
+end
+```
+
+We'll use the `Friends.Repo` module to query the database. We also tell this module to find its database configuration information in the `:friends` Elixir application.
+
+Next, we'll setup the `Friends.Repo` as a supervisor within our application's supervision tree in `lib/friends/application.ex`.  This will start the Ecto process when our application starts.
+
+```elixir
+  def start(_type, _args) do
+    # List all child processes to be supervised
+    children = [
+      Friends.Repo,
+    ]
+
+  ...
+```
+
+After that we'll need to add the following line to our `config/config.exs` file:
+```elixir
+config :friends, ecto_repos: [Friends.Repo]
+```
+
+This will allow our application to run ecto mix commands from the commandline.
+
+We're all done configuring the repository!  We can now create the database inside of postgres with this command:
+```shell
+$ mix ecto.create
+```
+
+Ecto will use the information in the `config/config.exs` file to determine how to connect to postgres and what name to give the database.  
+
+If you receive any errors, make sure that the configuration information is correct and that your instance of postgres is running.
+
+### Migrations
+
+To create and modify tables inside the postgres database, Ecto has *migrations*.  Each migration describes how the database tables should change.
+
+Our database doesn't have any tables yet.  To get started we'll create a migration that will create a *people* table.  Note that when naming a migration or table in ecto you should use the plural form of the word.
+
+We can use the following command to create a migration
+```shell
+$ mix ecto.gen.migration create_people
+```
+
+This will generate a new file in the `priv/repo/migrations` folder with the following contents.
+
+```elixir
+defmodule Friends.Repo.Migrations.CreatePeople do
+  use Ecto.Migration
+
+  def change do
+
+  end
+end
+```
+
+We can edit the change function to tell Ecto to create a table called *people* with a *name* column and *age* column.  You can also see that we define the data type of *name* to be a *string* and *age* to be an *int*.
+
+```elixir
+defmodule Friends.Repo.Migrations.CreatePeople do
+  use Ecto.Migration
+
+  def change do
+    create table(:people) do
+      add :name, :string
+      add :age, :integer
+    end
+  end
+end
+```
+
+If we run this migration, then a *people* table will be created with a *name* and *age* column:
+
+```shell
+mix ecto.migrate
+```
+
+### Schemas
+
+Schemas define how data from a database table or view maps to an elixir struct.  Tables are created using the plural form of the word but schemas should use the singular form.
+
+We'll create a *person* schema to match our people table.  This way, when we query our database, we can receive a collection of *person* elixir structs back.
+
+Let's create the schema at `lib/friends/person.ex`:
+
+```elixir
+defmodule Friends.Person do
```

Comment:
A lot of the time the naming pattern of modules matches the database table. There are times that this doesnt apply, lets say we had Teachers and Students, but all of them were `users`, but this goes into more complex use cases and relationships. Maybe make them match since this is the basics section? I personally think that makes the introduction here more straightforward. But I dont feel strongly, this is just observation from my end.

Reply:
To make sure I understand correctly, are you suggesting that `defmodule Friends.Person` would be `defmodule Friends.People` and file would be `people.ex` in this case?

Reply:
I would have this be `defmodule Friends.Person` with a filename `lib/friends/person.ex` with a table name `people` in the schema definition bit. The Schema is conventionally singular to avoid complexities of pluralization with ecto. Person vs People is an example of this.